nautobot/extras/signals.py

```diff
@@ -264,18 +266,32 @@ def git_repository_pre_delete(instance, **kwargs):
     # In fact, attempting to do so would cause database IntegrityErrors!
     job_result.use_job_logs_db = False
 
-    refresh_datasource_content("extras.gitrepository", instance, None, job_result, delete=True)
-
-    # In a distributed Nautobot deployment, each Django instance and/or worker instance may have its own clone
-    # of this repository; we need some way to ensure that all such clones are deleted.
-    # In the Celery worker case, we can broadcast a control message to all workers to do so:
-    app.control.broadcast("discard_git_repository", repository_slug=instance.slug)
-    # But we don't have an equivalent way to broadcast to any other Django instances.
-    # For now we just delete the one that we have locally and rely on other methods,
-    # such as the import_jobs_as_celery_tasks() signal that runs on server startup,
-    # to clean up other clones as they're encountered.
-    if os.path.isdir(instance.filesystem_path):
-        shutil.rmtree(instance.filesystem_path)
+    try:
+        refresh_datasource_content("extras.gitrepository", instance, None, job_result, delete=True)
+
+        # In a distributed Nautobot deployment, each Django instance and/or worker instance may have its own clone
+        # of this repository; we need some way to ensure that all such clones are deleted.
+        # In the Celery worker case, we can broadcast a control message to all workers to do so:
+        app.control.broadcast("discard_git_repository", repository_slug=instance.slug)
+        # But we don't have an equivalent way to broadcast to any other Django instances.
+        # For now we just delete the one that we have locally and rely on other methods,
+        # such as the import_jobs_as_celery_tasks() signal that runs on server startup,
+        # to clean up other clones as they're encountered.
+        if os.path.isdir(instance.filesystem_path):
+            shutil.rmtree(instance.filesystem_path)
+    except Exception as exc:
+        job_result.result = {
+            "exc_type": type(exc).__name__,
+            "exc_message": sanitize(str(exc)),
+        }
+        job_result.status = JobResultStatusChoices.STATUS_FAILURE
+        job_result.traceback = sanitize("".join(traceback.format_exception(type(exc), exc, exc.__traceback__)))
+        raise  # TODO: This presents in the UI as a 500 error
```

Comment:
Yeah, I don't think we should raise here. Just adding it to the jobresult is probably fine.